src/irc/core/irc.c

```diff
@@ -383,9 +383,45 @@ static void irc_server_event(IRC_SERVER_REC *server, const char *line,
 	g_free(event);
 }
 
+static char *get_tag_value(const char *tags, const char *tag)
+{
+	const char *val = NULL, *end;
+	int len = strlen(tag);
+
+	if (tags == NULL) {
+		return NULL;
+	}
+
+	if (strncmp(tags, tag, len) == 0 && tags[len] == '=') {
+		val = tags + len + 1;
+	} else {
+		char *needle = g_strconcat(";", tag, "=", NULL);
+		if ((val = strstr(tags, needle)) != NULL) {
+			val += len + 2;
+		}
+		g_free(needle);
+	}
+
+	if (val == NULL) {
+		return NULL;
+	}
+
+	if ((end = strchr(val, ';')) != NULL) {
+		return g_strndup(val, end - val);
+	} else {
+		return g_strdup(val);
+	}
+}
+
```

Comment:
I'd much prefer something that goes tag by tag, and puts every key-value pair it finds into a hash table. searching for ";key=" sounds like it will fail in a bunch of edge cases, and as we add support for more tags, this will just awkwardly re-scan the tag string several times.

Reply:
afaik the IRCv3 tag spec was written in such a way that there are no edge cases. the advantage of re-scanning is that we can just keep the current TAG-string parameter emitting code, otherwise we have to do something much more complicated.

Reply:
in any case, we (ideally) need a function that can be accessed from Perl that turns the existing tags string from irc_server_event_tags into the tag value

Reply:
>afaik the IRCv3 tag spec was written in such a way that there are no edge cases

Yeah, fair, not many. But value-less tags (`@key1=value1;key2;key3`) are definitely a thing.